replicators/src/mysql_connector/snapshot.rs

```diff
@@ -665,6 +680,54 @@ impl MySqlReplicator {
 
         Ok(())
     }
+
+    /// This function drops a single table from Noria.
+    ///
+    /// # Arguments
+    ///
+    /// * `noria`: The target ReadySet deployment
+    /// * `table`: The table to drop
+    async fn drop_single_table(
+        &mut self,
+        noria: &mut readyset_client::ReadySetHandle,
+        table: &Relation,
+    ) -> ReadySetResult<()> {
+        noria
+            .extend_recipe_no_leader_ready(ChangeList::from_change(
+                Change::Drop {
+                    name: table.clone(),
+                    if_exists: true,
+                },
+                Dialect::DEFAULT_MYSQL,
+            ))
+            .await?;
+        Ok(())
+    }
+
+    /// This function drops all known tables from Noria. This is used when we require a full
+    /// re-snapshot of the database.
+    ///
+    /// # Arguments
+    /// * `noria`: The target ReadySet deployment
+    async fn drop_all_tables(
+        &mut self,
+        noria: &mut readyset_client::ReadySetHandle,
+    ) -> ReadySetResult<()> {
+        let replication_offsets = noria.replication_offsets().await?;
+        let mut tx = self.pool.start_transaction(tx_opts()).await?;
+        let all_tables = get_table_list(&mut tx, TableKind::BaseTable).await?;
+        for (db, table) in all_tables.iter() {
+            let table = Relation {
+                schema: Some(SqlIdentifier::from(db.clone())),
+                name: SqlIdentifier::from(table.clone()),
+            };
+            if replication_offsets.has_table(&table) {
+                self.drop_single_table(noria, &table).await?;
```

Comment:
hmm... I kinda feel like instead of sending a bunch of `extend_recipe` RPCs to drop each table one by one, it's going to be cleaner, faster, and less prone to race conditions to send a single `extend_recipe` with a `Change::Drop` change for each table to drop. WDYT?

Reply:
I agree. I had originally done this with a vector of changes, but later I thought in splitting the drop on a single table function to be reused later. But I agree it will have a better performance with a single RPC call.